plugins/win-wasapi/audio-notify-client.cpp

```diff
@@ -0,0 +1,66 @@
+#include "audio-notify-client.h"
+#include <windows.h>
+#include <assert.h>
+
+WASAPINotify::WASAPINotify(IWASNotifyCallback *cb_) : cb(cb_)
+{
+	assert(cb);
+}
+
+STDMETHODIMP_(ULONG) WASAPINotify::AddRef()
+{
+	return InterlockedIncrement(&refs);
+}
+
+STDMETHODIMP_(ULONG) STDMETHODCALLTYPE WASAPINotify::Release()
+{
+	long temp = InterlockedDecrement(&refs);
+	if (temp == 0)
+		delete this;
+
+	return temp;
+}
+
+STDMETHODIMP WASAPINotify::QueryInterface(REFIID riid, void **ptr)
+{
+	if (riid == IID_IUnknown) {
+		*ptr = (IUnknown *)this;
+	} else if (riid == __uuidof(IMMNotificationClient)) {
+		*ptr = (IMMNotificationClient *)this;
+	} else {
+		*ptr = nullptr;
+		return E_NOINTERFACE;
+	}
+
+	InterlockedIncrement(&refs);
+	return S_OK;
+}
+
+STDMETHODIMP WASAPINotify::OnDeviceAdded(LPCWSTR)
```

Comment:
If this is inherits a callback class, is having these empty callbacks necessary versus just not defining them?